src/main/java/io/molnarsandor/trelloclone/project/ProjectService.java

```diff
@@ -0,0 +1,141 @@
+            backlogEntity.setProjectIdentifier(projectEntity.getProjectIdentifier().toUpperCase());
+        }
+
+        if (projectEntity.getId() != null) {
+            projectEntity.setBacklog(backlogRepository.findByProjectIdentifierIgnoreCase(projectEntity.getProjectIdentifier().toUpperCase()));
+        }
+
+        return modelConverter.projectEntityToDto(
+                projectRepository.save(projectEntity));
+
+    }
+
+    public List<ProjectDTO> findAllProject(String username) {
+
+        List<ProjectEntity> byLeader = projectRepository.findAllByProjectLeader(username);
+
+        CollaboratorEntity collaboratorEntity = collaboratorRepository.findByEmail(username);
+
+        List<ProjectEntity> byCollaborator = projectRepository.findAllByCollaborators(collaboratorEntity);
+
+        byLeader.addAll(byCollaborator);
+
+        return modelConverter.projectEntityListToDto(byLeader);
+    }
+
+    public DeleteDTO deleteProjectByIdentifier(String projectId, String username) {
+
+        projectRepository.delete(findProjectByIdentifier(projectId, username));
+
+        return new DeleteDTO("Project " + projectId + " deleted");
+    }
+
+    // == PRIVATE METHODS ==
+    private void validateProject(ProjectEntity projectEntity, String projectIdentifier, String username) {
+        if (projectEntity != null && !projectEntity.getProjectLeader().equals(username)) {
+            throw new ProjectNotFoundException("Project not found in your account");
+        } else if (projectEntity == null) {
+            throw new ProjectNotFoundException("Project with ID: '" + projectIdentifier + "' cannot be updated because it does not exist");
+        }
+    }
+
+    private void validateProjectWithCollaborators(ProjectEntity projectEntity, String projectId, String username) {
+        if (projectEntity == null) {
+            throw new ProjectNotFoundException("Project ID '" + projectId + "' does not exists");
+        }
+
+        Set<CollaboratorEntity> collaboratorEntities = projectEntity.getCollaborators();
+
+        Predicate<CollaboratorEntity> collaboratorPredicate = collaborator ->
+                collaborator.getProjectIdentifier().equalsIgnoreCase(projectId) &&
+                        collaborator.getEmail().equals(username);
+
+
+        boolean collaboratorExists = false;
+        if (!collaboratorEntities.isEmpty()) {
+            collaboratorExists = collaboratorEntities
+                    .stream()
+                    .anyMatch(collaboratorPredicate);
+        }
+
+        if (!projectEntity.getProjectLeader().equals(username) && !collaboratorExists) {
+            throw new ProjectNotFoundException("Project not found in your account");
+        }
+    }
+
+    private ProjectEntity getProjectById(String projectId, String username) {
+        ProjectEntity projectEntity = projectRepository.findByProjectIdentifierIgnoreCase(projectId);
+
+        validateProjectWithCollaborators(projectEntity, projectId, username);
```

Comment:
Ez a metódus 2 dolgot is csinál, felszedi a Project-et az adatbázisból, illetve még üzleti logikát is validál, ami nem jó practice. Ez a metódus csak adja vissza a Project entitást, a validációt meg talán tedd publikussá. Illetve én át is nevezném, a `validate` szerintem elég általános elnevezés ellentétben azzal amit csinál: megvizsgálja, hogy a User-é e a Project / van-e joga elvégezni valamit - ehelyett pl. `userCanManageProject` vagy valami hasonló.